### ISSUE TITLE
Mouse events are does not seem to work on Markers

### ISSUE THREAD

**Issue by vaasu070** [airbnb/visx#868]
Hi , I have been using VX library for a while now I am liking it. 

For markers I am using `MarkerCircle `on `LinePath `elements. The tooltip is expected to open when I hover anywhere on the plotted line, but it only opens when mouse is hovered  over the line and  not when hovered over markers. I have defined mouse events for `MarkerCircle `separately , still it does not seem to work.

codesandbox link for the same [https://codesandbox.io/s/cool-http-xyejr?file=/src/App.js:1786-1798]

And In case If I am plotting 100 lines, is It possible to change the marker fill colors dynamically (like passing color as props to `MarkerCircle  ` )wihout adding 100  `MarkerCircle ` element inside SVG element?

**Comment by vaasu070** [comment]
Is it at least possible to change the z-index of markers so that markers come below the line and mouse events are not ignored?

**Comment by hshoff** [comment]
No, It is not possible because there is no z-index in svg.

It's possible with the `<circle />` approach but you would need to render the cicles before the line to mimic z-index (svg follows painter model).

**Comment by williaster** [comment]
@vaasu070 you could also consider using a `voronoi` (`@visx/voronoi`) to map mouse events to the nearest point. This is a common approach so a user's mouse doesn't have to be exactly on a line or small point to trigger the event. See http://airbnb.io/visx/voronoi for an example

**Comment by vaasu070** [comment]
@hshoff , Yes I have considered your suggestion. As I am using large set of data points, performance was being affected by it. Many Thanks

**Comment by vaasu070** [comment]
Strangely specifically for some lines , mouse-events like `mouseEnter `, `onClick` are being triggered even before mouse pointer hovers over the lines. As a result of this, tooltip opens early. But for other lines ,this is working completely fine.  Why this could be happening?

![image](https://user-images.githubusercontent.com/33760706/96213192-05197e80-0f96-11eb-9af6-f9a5f00aeb2a.png)

**Comment by vaasu070** [comment]
> No, it is not possible because there is no z-index in svg.
> 
> It's possible with the `<circle />` approach but you would need to render the cicles before the line to mimic z-index (svg follows painter model).

@hshoff , Yes I have considered your suggestion. As I am using large set of data points, performance was being affected by it so I am sticking with previous implementation itself. Many Thanks

**Comment by vaasu070** [comment]
> Strangely specifically for some lines , mouse-events like `mouseEnter `, `onClick` are being triggered even before mouse pointer hovers over the lines. As a result of this, tooltip opens early. But for other lines ,this is working completely fine. Why this could be happening?
> 
> ![image](https://user-images.githubusercontent.com/33760706/96213776-67bf4a00-0f97-11eb-868a-3b2b438273b8.png)

Any help on this?

**Comment by williaster** [comment]
Hey @vaasu070 we're not going to be able to help debug this without a codesandbox to reproduce. It could be any number of things.

**Comment by vaasu070** [comment]
> Hey @vaasu070 we're not going to be able to help debug this without a codesandbox to reproduce. It could be any number of things.

[https://codesandbox.io/s/headless-haze-zkekj?file=/src/App.js] code sandbox link has been added here.

**Comment by hshoff** [comment]
Answered here: https://github.com/airbnb/visx/issues/897
